src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -186,7 +186,8 @@ private static void setupSignalHandler(RubyContext context) {
         try (Pointer structSigAction = context.getNativePlatform().createSigAction(nfi.asPointer(abs))) {
             int result = (int) sigaction.call(Signal.SIGVTALRM.intValue(), structSigAction.getAddress(), 0L);
             if (result != 0) {
-                throw new UnsupportedOperationException("sigaction() failed: errno=" + context.getNativePlatform().getPosix().errno());
+                // TODO (eregon, 24 Nov. 2017): we should show the NFI errno here.
+                throw new UnsupportedOperationException("sigaction() failed");
```

Comment:
Can you elaborate on what was wrong about the previous behavior?

Reply:
It was reading the JNR-private/virtual errno but this is a NFI call which doesn't set that errno (currently NFI sets the real system errno but that's not convenient to get here, will soon be much easier).